include/zephyr/net/net_l2_connectivity.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for managing L2 layers that have separated post-admin-up connect
+ *	  or associate phases
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_L2_CONNECTIVITY_H_
+#define ZEPHYR_INCLUDE_NET_L2_CONNECTIVITY_H_
+
+#include <zephyr/device.h>
+#include <zephyr/net/net_if.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief Network Layer 2 connectivity management abstraction layer
+ * @defgroup net_l2_connectivity Network L2 Connectivity Abstraction Layer
+ * @ingroup networking
+ * @{
+ */
+
+/**
+ * @brief Network L2 connectivity API structure
+ *
+ * Used to provide an interface to connect parameters and procedures
+ */
+struct net_l2_connectivity_api {
+	int (*connect)(struct net_if *iface);
+	int (*disconnect)(struct net_if *iface);
+
+	/* Note to reviewers:
+	 * Technically, as long as each iface provides a means by which to set
+	 * association/connection parameters, these get/set_opt funcs are not needed.
+	 *
+	 * However, it can be argued that they should still exist, since they establish a
+	 * standardized pattern for doing that.
+	 *
+	 * The question, then, is: Why limit these to specifically connectivity?
```

Comment:
Personally, I'd stick to connectivity only. Those sort of functions are already pretty obfuscated by their nature, if they get too much responsibility (i. e. configuring every interface param one could imagine) it could get quite confusing.

Reply:
Plus, if we don't have any particular use case on the horizon that would justify this we should not add it.

https://en.wikipedia.org/wiki/You_aren't_gonna_need_it

Reply:
I'm somewhat relieved you both feel this way, after having sat with the question for a few days myself, this is where my opinion landed too